### PR TITLE
Finalize version 1.3 and publish to the public

### PR DESCRIPTION
**Pull request by Ubuntufanboy**
Version 1.3 removes the need to write in the size of your terminal. Version 1.3 also includes many features to stop common problems. Also, our launcher.py file was born to help users run the code with ease

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -1,4 +1,9 @@
+sudo apt-get update
+sudo apt-get upgrade
+apt install python3-brotli -y
 sudo apt install yt-dlp -y
 sudo apt install ffmpeg -y
 pip3 install pillow
 pip3 install playsound
+pip3 install tqdm
+echo 'All done! To run the code, type python3 launcher.py'
```
Comment (Debianfanboy) [inline install.sh:9]: Should have new line here smh

File: launcher.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    length = sum(1 for x in p.glob('*') if x.is_file())
+    if length >= 3284:
+        print("Already exists")
+        skip = True
+    else:
+        print(f"Expected 3285 files. Got {length}")
+        print("Too few files in directory")
+else:
+    print("Directory does not exist")
+    os.mkdir("img")
```
Comment (Debianfanboy) [inline launcher.py:25]: We should use try block here

File: main.py [hunk 3]
```diff
@@ -1,167 +1,126 @@
...
-from time import sleep
-from os import system
+import numpy as np
+import os
+import time
 from silver import Silver
 from threading import Thread
+from tqdm import tqdm
+from pathlib import Path
+def process(img: np.ndarray) -> str:
```
Comment (Debianfanboy) [inline main.py:8]: Real nice job speeding up code by ~1.5x here

File: main.py [hunk 4]
```diff
@@ -1,167 +1,126 @@
...
+    os.chdir("processed")
+    place = f"{x}x{y}"
+    os.chdir(place)
+    Silver.play(f"{current}/badapple.mp3")
+    for i in range(1,3286):
+        filename = str(i) + ".txt"
+        with open(filename) as f:
+            frame = f.read()
+            print(frame)
+            time.sleep(0.0625)
```
Comment (Debianfanboy) [inline main.py:126]: Should have another line at end. Also should have a comment explaining that sometimes the timing is off

File: README.md [hunk 5]
```diff
@@ -20,7 +20,7 @@ Thrid, run this command ``chmod +x install.sh``
 
 Now you can run ``bash ./install.sh``
 
-Next run, ``chmod +x launcher.sh && bash ./launcher.sh``
+Next run, ``python3 launcher.py``
```
Comment (Debianfanboy) [inline README.md:23]: Docs is good

File: convert.py [hunk 6]
```diff
@@ -1,19 +1,11 @@
 # ./img/output000001.png
 
-from os import system
+from tqdm import tqdm
+import os, glob
 from PIL import Image
+from pathlib import Path
 num = 1
 # list of all files
-print("Hey! This part of the code breaks a lot so listen up! *press enter to continue*")
```
Comment (Debianfanboy) [inline convert.py:7]: thank god this mess is gone

File: install.sh (hunk 1 above)
Comment (Debianfanboy) [inline install.sh:9]: wtf add new line idiot

File: launcher.py (hunk 2 above)
Comment (Debianfanboy) [reply launcher.py:25]: yeah add try block

File: main.py (hunk 3 above)
Comment (Debianfanboy) [reply main.py:8]: good stuff

File: main.py [hunk 7]
```diff
@@ -1,167 +1,126 @@
...
-                else:
-                    frame += " "
-            w1.append(frame)
+    for i in tqdm(range(1,1000)):
+        try:
+            with Image.open(f"{current}/converted/{x}x{y}/new{i}.png") as img:
+                img = img.getdata(0)
+                img = np.array(img)
+                w1.append(process(img))
+        except:
```
Comment (Debianfanboy) [inline main.py:31]: naw wtf is this raw except

File: main.py [hunk 8]
```diff
@@ -1,167 +1,126 @@
...
+global y
+x = os.get_terminal_size().columns
+y = os.get_terminal_size().lines
+
+current = os.getcwd()
+p = Path(f"{current}/processed/{x}x{y}")
+images = Path(f"{current}/converted/{x}x{y}")
+
+if images.exists():
+    pass
```
Comment (Debianfanboy) [inline main.py:69]: why... why the pass in the if...

File: main.py [hunk 9]
```diff
@@ -1,167 +1,126 @@
...
 k = Thread(target=wk4)
 
-f.start()
-u.start()
-c.start()
-k.start()
+print("Checking terminal size! Please do not change to terminal size")
+time.sleep(1)
+
+global x
```
Comment (Debianfanboy) [inline main.py:59]: pretty sure the global is not needed

**Review by Debianfanboy — CHANGES_REQUESTED** [review]
Please fix these things. I wrote comments on the things needed to be fixed. Once done, please test code thoroughly. After that you are good to merge

**Comment by Ubuntufanboy** [comment]
Fixed all requested changes!

**Comment by Debianfanboy** [comment]
LGTM! Ready to merge and publish!

**Comment by Ubuntufanboy** [comment]
Thanks, Merging

**Pull request merged**